bindings/java/src/test/java/org/apache/opendal/test/behavior/AbstractBehaviorTest.java

```diff
@@ -164,6 +164,45 @@ public void testAppendCreateAppend() {
         }
     }
 
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    @Nested
+    class AsyncCreateDirTest {
+        @BeforeAll
+        public void precondition() {
+            final Capability capability = operator.info.fullCapability;
+            assumeTrue(capability.createDir);
+        }
+
+        /**
+         * Create dir with dir path should succeed.
+         */
+        @Test
+        public void testCreateDir() {
+            final String path = String.format("%s/", UUID.randomUUID().toString());
+            operator.createDir(path).join();
+
+            try (final Metadata meta = operator.stat(path).join()) {
+                assertThat(meta.isFile()).isFalse();
+            }
+            operator.delete(path).join();
+        }
+
+        /**
+         * Create dir on existing dir should succeed.
+         */
+        @Test
+        public void testCreateDirExisting() {
```

Comment:
Binding doesn't need to test behavior like this, they are covered by rust core. Unless binding has its own internal logic.

Reply:
Such tests align the behavior from bindings to the core - if the core changes (signature), bindings fail and get aware.

Otherwise, we almost need not any test which is different from the previous Cucumber tests solution.

Reply:
You may take Kafka's C# client which is a binding of C++ client as an example: https://github.com/confluentinc/confluent-kafka-dotnet/tree/master/test

Behavior tests can also serve as examples for users on how to use these APIs and ensure that these APIs are properly delivered by the binding.

Reply:
Take this patch as a real example - how can I know I write the binding method correct? Anyway I have to write some demos to test it out. And these are what they are.